### PR TITLE
bpo-43422: Revert _decimal C API addition

### PR DESCRIPTION
**Pull request by pitrou**
Stefan Krah requested the reversal of these (unreleased) changes, quoting him:
> The capsule API does not meet my testing standards, since I've focused
on the upstream mpdecimal in the last couple of months.
> Additionally, I'd like to refine the API, perhaps together with the
Arrow community.

https://bugs.python.org/issue43422

### CODE REVIEW

**Comment by erlend-aasland** [comment]
Shouldn't this require a NEWS item? #21519 had one, and the API has been present in all the 3.10 alphas released so far.

File: Modules/_decimal/_decimal.c [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2008-2020 Stefan Krah. All rights reserved.
+ * Copyright (c) 2008-2012 Stefan Krah. All rights reserved.
```
Comment (mdickinson) [inline Modules/_decimal/_decimal.c:2]: This change seems odd; is this definitely correct? (Should "2012" have been "2021" here?)

I don't think it's terribly consequential, and I'm happy for the change to stay if that keeps us compatible with upstream.

File: Modules/_decimal/tests/deccheck.py [hunk 2]
```diff
@@ -47,18 +47,14 @@
...
 from _pydecimal import _dec_from_triple
 
-from _testcapi import decimal_as_triple
-from _testcapi import decimal_from_triple
-
-import _decimal as C
-import _pydecimal as P
-
-
+C = import_fresh_module('decimal', fresh=['_decimal'])
```
Comment (mdickinson) [inline Modules/_decimal/tests/deccheck.py:56]: Are these changes related to the purpose of the PR, or just incidental?

**Review by mdickinson — APPROVED** [review]
LGTM. A couple of questions, but nothing show-stopping.

**Comment by pitrou** [comment]
For the record, this PR is straight from Stefan's own branch here: https://github.com/python/cpython/compare/master...skrah:revert_decimal_capsule_api
I'm reluctant to do any changes that aren't necessary for inclusion.

**Comment by pitrou** [comment]
I'll add a NEWS item soon.

**Comment by pitrou** [comment]
Since this PR was created from Stefan's fork, I've opened #24960 from my fork in order to add a NEWS item. Closing this one.

**Pull request closed without merging**